access/types/forms.py

```diff
@@ -522,27 +546,40 @@ def row_options(self, configuration, row):
     def grade_checkbox(self, configuration, value, hints=None):
         hints = hints or []
         correct_count = 0
-
-        # All correct required if any configured
+        wrong_answers = 0
+        non_neutral_count = 0
+        # All correct required if any configured and partial-points is not set
         correct = True
         i = 0
         for opt in configuration.get("options", []):
             name = self.option_name(i, opt)
-            if opt.get("correct", False):
+            correct_answer = opt.get("correct", False)
+            # correct_answer may be boolean or string "neutral"
+            if correct_answer is True:
                 correct_count += 1
+                non_neutral_count += 1
                 if name not in value:
+                    wrong_answers += 1
                     correct = False
                     self.append_hint(hints, opt)
-            elif not value is None and name in value:
-                correct = False
-                self.append_hint(hints, opt)
+            elif correct_answer is False:
+                non_neutral_count += 1
+                if value is not None and name in value:
+                    correct = False
+                    wrong_answers += 1
+                    self.append_hint(hints, opt)
             i += 1
 
+        # If partial_points are set, correct is scaled to 0-1 float, to be
+        # factored to max points.
```

Comment:
I'm not sure the word "factored" is used correctly in the comment. For example, arent't 2 and 6 factors to 12 because `2 * 6 = 12`? I was thinking that you are actually scaling the points with the float number.